src/main/rx/flysky.c

```diff
@@ -0,0 +1,207 @@
+    {0x82, 0x32, 0x6e, 0x1e, 0x78, 0x28, 0xa0, 0x50, 0x64, 0x14, 0x8c, 0x3c, 0x5a, 0x0a, 0x96, 0x46},
+    {0x46, 0x96, 0x0a, 0x5a, 0x50, 0xa0, 0x3c, 0x8c, 0x28, 0x78, 0x1e, 0x6e, 0x32, 0x82, 0x14, 0x64},
+    {0x64, 0x14, 0x82, 0x32, 0x6e, 0x1e, 0x78, 0x28, 0x8c, 0x3c, 0xa0, 0x50, 0x5a, 0x0a, 0x96, 0x46}
+};
+
+
+typedef struct __attribute__((packed))
+{
+    uint8_t  type;
+    uint32_t txId;
+    uint8_t  data[FLYSKY_CHANNEL_COUNT*2];
+} rcDataPkt_t;
+
+
+static uint32_t txId;
+static uint8_t  channelRow;
+static uint8_t  channelOffset;
+static bool binded;
+
+
+static inline void setTxId (uint32_t txid);
+static inline uint8_t getNextChannel (void);
+
+
+void FlySkyInit(const struct rxConfig_s *rxConfig, struct rxRuntimeConfig_s *rxRuntimeConfig)
+{
+    if (rxRuntimeConfig) { rxRuntimeConfig->channelCount = FLYSKY_CHANNEL_COUNT; }
+
+    IO_t bindIO = IOGetByTag( IO_TAG(RX_FLYSKY_BIND) );
+    IOInit(bindIO, OWNER_RX_SPI_CS, 0);
+    IOConfigGPIO(bindIO, IOCFG_IPU);
+
+    A7105Init(0x5475c52A);
+    A7105Config(A7105Regs, sizeof(A7105Regs));
+
+    if ( !IORead(bindIO) || rxConfig->rx_spi_id == 0 )
+    {
+        binded = false;
+        A7105WriteReg(A7105_0F_CHANNEL, 0);            /* binding listen on channel 0 */
+        A7105Strobe(A7105_RX);
+    }
+    else
+    {
+        binded = true;
+        setTxId(rxConfig->rx_spi_id);
+    }
+}
+
+
+void FlySkySetRcDataFromPayload(uint16_t *rcData, const uint8_t *payload)
+{
+    if (rcData && payload)
+    {
+        for (uint8_t i = 0; i < FLYSKY_CHANNEL_COUNT; i++)
+        {
+            rcData[i] = payload[2 * i + 1] << 8 | payload [2 * i + 0];
+        }
+    }
+}
+
+
+rx_spi_received_e  FlySkyDataReceived(uint8_t *payload)
+{
+    static uint32_t timeout = 1600; /* us */
+    static uint32_t timeLastPacket = 0;
+    static uint16_t countTimeout = 0;
+    static uint8_t packet[FLYSKY_PAYLOAD_SIZE];
+    rx_spi_received_e result = RX_SPI_RECEIVED_NONE;
+    uint32_t time;
+
+    if (A7105RxTxFinished(&time))
+    {
+        uint8_t modeReg = A7105ReadReg(A7105_00_MODE);
+
+        if ((modeReg & (A7105_MODE_CRCF|A7105_MODE_TRER)) == 0 )    /* CRC pass */
+        {
+            A7105ReadFIFO(packet, FLYSKY_PAYLOAD_SIZE);
+            rcDataPkt_t *rcPacket = (rcDataPkt_t *) packet;
+            if (binded)
+            {
+                if (rcPacket->txId == txId)
+                {
+                    timeLastPacket = time;
+                    A7105WriteReg(A7105_0F_CHANNEL, getNextChannel());
+                    countTimeout = 0;
+                    timeout = 1600; /* us */
+                    if (payload) { memcpy(payload, rcPacket->data, 2*FLYSKY_CHANNEL_COUNT); }
+                    result = RX_SPI_RECEIVED_DATA;
+                }
+            }
+            else
+            {
+                if (rcPacket->type == 0xAA)
+                {
+                    setTxId(rcPacket->txId);
+                    rxConfigMutable()->rx_spi_id = txId;

```

Comment:
This use of  `rxConfigMutable()->rx_spi_id` is inconsistent with using `flySkyA2Config_t`. Better use the same config structure / parameter group for both sub-protocols.